pep-0639.rst

```diff
@@ -1,543 +1,2208 @@
+  Examples: Wheel project, PEP 517 backends via ``build`` or other
+  package-developer-facing frontends, calling ``setup.py`` directly.
+
+  An **install tool** is a packaging tool used to install a source or built
+  distribution in a target environment. Examples include the PyPA pip and
+  ``installer`` projects.
+
+  A **publishing tool** is a packaging tool used to upload distribution
+  archives to a package index, such as Twine for PyPI.
+
+**Wheel Format** *(Short: wheel, Rel: Wheel project)*
+  Here, **wheel**, the standard built distribution format introduced in PEP 427
+  and `specified by PyPA <#wheelspec_>`_, will be referred to in lowercase,
+  while the `Wheel project <#wheelproject_>`_, its reference implementation,
+  will be referred to as **Wheel** in Title Case.
+
+
+Specification
+=============
+
+The changes necessary to implement the improved license handling outlined in
+this PEP include those in both
+`distribution package metadata <Core metadata_>`_, as defined in the
+`core metadata specification <#coremetadataspec_>`_, and
+`author-provided project source metadata <Project source metadata_>`_, as
+originally defined in PEP 621.
+
+Further, `minor additions <License files in project formats_>`_ to the
+source distribution (sdist), built distribution (wheel) and installed project
+specifications will help document and clarify the already allowed,
+now formally standardized behavior in these respects.
+Finally, `guidance is established <Converting legacy metadata_>`_
+for tools handling and converting legacy license metadata to license
+expressions, to ensure the results are consistent, correct and unambiguous.
+
+
+Core metadata
+-------------
+
+The `PyPA Core Metadata specification <#coremetadataspec_>`_ defines the names
+and semantics of each of the supported fields in the distribution metadata of
+Python distribution packages and installed projects.
+
+This PEP `adds <Add License-Expression field_>`_ the
+``License-Expression`` field,
+`adds <Add License-File field_>`_ the ``License-File`` field,
+`deprecates <Deprecate License field_>`_ the ``License`` field,
+and `deprecates <Deprecate license classifiers_>`_ the license classifiers
+in the ``Classifier`` field.
+
+The error and warning guidance in this section applies to build and
+publishing tools; user-facing install tools MAY be more lenient than
+mentioned here when encountering malformed metadata
+that does not conform to this specification.
+
+As it adds new fields, this PEP updates the core metadata to version 2.3.
+
+
+Add ``License-Expression`` field
+''''''''''''''''''''''''''''''''
+
+The ``License-Expression`` optional field is specified to contain a text string
+that is a valid SPDX license expression, as defined herein.
+
+Publishing tools SHOULD issue an informational warning if this field is
+missing, and MAY raise an error. Build tools MAY issue a similar warning,
+but MUST NOT raise an error.
```

Comment:
Good point. I would like that as well; I was just trying to stick to the intent of the existing language in the PEP where practical. I guess this was implicitly "by default", but we could clarify it like so:

```suggestion
missing, and MAY raise an error. Build tools MAY issue a similar warning,
but MUST NOT raise an error by default, without being configured otherwise.
```

*However*, I think a better alternative, instead of mentioning it under every validation provision, is having a sentence at the top of the spec which states the following:

```rst
The validation guidance below is for tools' default behavior;
they MAY operate more strictly if users explicitly configure them to do so,
such as by a CLI flag or a configuration option.
```

I've thus gone ahead with the latter.

Reply:
Sounds good 👍

Reply:
Thanks!